I rebaselined on develop but now the variant doesn't appear to work the same way

If I revert f4e66b306e7bfe96ac57258d3cfb96b63ae553bc (https://github.com/wortiz/spack/tree/trilinos-x11-revert) it appears to work again

Without the revert I end up with a link error. I couldn't spot what was causing this issue quickly, nothing obvious popped out at me.

        1877    /usr/bin/ld: CMakeFiles/SEACASSupes_exttest.dir/exttest.f.o: in function `MAIN__':
      >> 1878    /tmp/wortiz/spack-stage/spack-stage-trilinos-13.0.1-oevihz2c7yehcbnqfhpty5cpwdvl7j5s/spack-src/packages/seacas/libraries/supes/exttes
                 t.f:12: undefined reference to `excpus_'
      >> 1879    /usr/bin/ld: /tmp/wortiz/spack-stage/spack-stage-trilinos-13.0.1-oevihz2c7yehcbnqfhpty5cpwdvl7j5s/spack-src/packages/seacas/libraries
                 /supes/exttest.f:21: undefined reference to `exupcs_'
      >> 1880    /usr/bin/ld: /tmp/wortiz/spack-stage/spack-stage-trilinos-13.0.1-oevihz2c7yehcbnqfhpty5cpwdvl7j5s/spack-src/packages/seacas/libraries
                 /supes/exttest.f:24: undefined reference to `exdate_'
      >> 1881    /usr/bin/ld: /tmp/wortiz/spack-stage/spack-stage-trilinos-13.0.1-oevihz2c7yehcbnqfhpty5cpwdvl7j5s/spack-src/packages/seacas/libraries
                 /supes/exttest.f:27: undefined reference to `extime_'
      >> 1882    /usr/bin/ld: /tmp/wortiz/spack-stage/spack-stage-trilinos-13.0.1-oevihz2c7yehcbnqfhpty5cpwdvl7j5s/spack-src/packages/seacas/libraries
                 /supes/exttest.f:30: undefined reference to `exname_'
      >> 1883    /usr/bin/ld: /tmp/wortiz/spack-stage/spack-stage-trilinos-13.0.1-oevihz2c7yehcbnqfhpty5cpwdvl7j5s/spack-src/packages/seacas/libraries
                 /supes/exttest.f:37: undefined reference to `exname_'
      >> 1884    /usr/bin/ld: /tmp/wortiz/spack-stage/spack-stage-trilinos-13.0.1-oevihz2c7yehcbnqfhpty5cpwdvl7j5s/spack-src/packages/seacas/libraries
                 /supes/exttest.f:44: undefined reference to `exname_'
      >> 1885    /usr/bin/ld: /tmp/wortiz/spack-stage/spack-stage-trilinos-13.0.1-oevihz2c7yehcbnqfhpty5cpwdvl7j5s/spack-src/packages/seacas/libraries
                 /supes/exttest.f:51: undefined reference to `exparm_'
      >> 1886    /usr/bin/ld: /tmp/wortiz/spack-stage/spack-stage-trilinos-13.0.1-oevihz2c7yehcbnqfhpty5cpwdvl7j5s/spack-src/packages/seacas/libraries
                 /supes/exttest.f:55: undefined reference to `exmemy_'
      >> 1887    /usr/bin/ld: /tmp/wortiz/spack-stage/spack-stage-trilinos-13.0.1-oevihz2c7yehcbnqfhpty5cpwdvl7j5s/spack-src/packages/seacas/libraries
                 /supes/exttest.f:59: undefined reference to `exmemy_'
      >> 1888    /usr/bin/ld: /tmp/wortiz/spack-stage/spack-stage-trilinos-13.0.1-oevihz2c7yehcbnqfhpty5cpwdvl7j5s/spack-src/packages/seacas/libraries
                 /supes/exttest.f:62: undefined reference to `ixlnum_'
      >> 1889    /usr/bin/ld: /tmp/wortiz/spack-stage/spack-stage-trilinos-13.0.1-oevihz2c7yehcbnqfhpty5cpwdvl7j5s/spack-src/packages/seacas/libraries
                 /supes/exttest.f:62: undefined reference to `ixlnum_'
      >> 1890    /usr/bin/ld: /tmp/wortiz/spack-stage/spack-stage-trilinos-13.0.1-oevihz2c7yehcbnqfhpty5cpwdvl7j5s/spack-src/packages/seacas/libraries
                 /supes/exttest.f:65: undefined reference to `ixlchr_'
      >> 1891    /usr/bin/ld: /tmp/wortiz/spack-stage/spack-stage-trilinos-13.0.1-oevihz2c7yehcbnqfhpty5cpwdvl7j5s/spack-src/packages/seacas/libraries
                 /supes/exttest.f:65: undefined reference to `ixlchr_'
      >> 1892    /usr/bin/ld: /tmp/wortiz/spack-stage/spack-stage-trilinos-13.0.1-oevihz2c7yehcbnqfhpty5cpwdvl7j5s/spack-src/packages/seacas/libraries
                 /supes/exttest.f:73: undefined reference to `excpus_'
      >> 1893    /usr/bin/ld: libsupes.so.13.0: undefined reference to `exread_'

--- a/var/spack/repos/builtin/packages/trilinos/package.py
+++ b/var/spack/repos/builtin/packages/trilinos/package.py
@@ -3,7 +3,6 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
-import os
 import sys
 
 from spack import *
@@ -683,16 +682,16 @@ class Trilinos(CMakePackage, CudaPackage):
         # ################# System-specific ######################
 
         # Fortran lib (assumes clang is built with gfortran!)
-        if ('+fortran +mpi' in spec
-                and spec.compiler.name in ['gcc', 'clang', 'apple-clang']):
-            mpifc = Executable(spec['mpi'].mpifc)
-            libext = 'dylib' if sys.platform == 'darwin' else 'so'
-            libgfortran_dir = os.path.dirname(
-                mpifc('--print-file-name', 'libgfortran.%s' % libext, output=str))
-            options.append(define(
-                'Trilinos_EXTRA_LINK_FLAGS',
-                '-L%s/ -lgfortran' % (libgfortran_dir),
-            ))
+        if ('+fortran' in spec and spec.compiler.name in ['gcc', 'clang', 'apple-clang']):
+            fc = Executable(spec['mpi'].mpifc) if ('+mpi' in spec) else Executable(spack_fc)
+            sharedlibext = 'dylib' if sys.platform == 'darwin' else 'so'
+            libgfortran = fc('--print-file-name',
+                'libgfortran.%s' % sharedlibext, output=str).strip()
+            # if libgfortran is equal to "libgfortran.<libext>" then
+            # print-file-name failed, use static library instead
+            if (libgfortran == ('libgfortran.%s' %  sharedlibext)):
+                libgfortran = fc('--print-file-name', 'libgfortran.a', output=str).strip()
+            options.append(define('Trilinos_EXTRA_LINK_FLAGS', libgfortran))
 
         if sys.platform == 'darwin' and macos_version() >= Version('10.12'):
             # use @rpath on Sierra due to limit of dynamic loader

--- a/var/spack/repos/builtin/packages/trilinos/package.py
+++ b/var/spack/repos/builtin/packages/trilinos/package.py
@@ -3,6 +3,7 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+import os
 import sys
 
 from spack import *
@@ -83,6 +84,7 @@ class Trilinos(CMakePackage, CudaPackage):
     variant('superlu-dist', default=False, description='Compile with SuperluDist solvers')
     variant('superlu',      default=False, description='Compile with SuperLU solvers')
     variant('strumpack',    default=False, description='Compile with STRUMPACK solvers')
+    variant('x11',          default=False, description='Compile with X11 when +exodus')
     variant('zlib',         default=False, description='Compile with zlib')
 
     # Package options (alphabet order)
@@ -319,7 +321,7 @@ class Trilinos(CMakePackage, CudaPackage):
 
     # Variant requirements from packages
     depends_on('metis', when='+zoltan')
-    depends_on('libx11', when='+exodus')
+    depends_on('libx11', when='+x11')
     depends_on('matio', when='+exodus')
     depends_on('netcdf-c', when="+exodus")
     depends_on('netcdf-c+mpi+parallel-netcdf', when="+exodus+mpi@12.12.1:")
@@ -684,10 +686,12 @@ class Trilinos(CMakePackage, CudaPackage):
         if ('+fortran +mpi' in spec
                 and spec.compiler.name in ['gcc', 'clang', 'apple-clang']):
             mpifc = Executable(spec['mpi'].mpifc)
-            libgfortran = mpifc('--print-file-name', 'libgfortran.a', output=str)
+            libext = 'dylib' if sys.platform == 'darwin' else 'so'
+            libgfortran_dir = os.path.dirname(
+                mpifc('--print-file-name', 'libgfortran.%s' % libext, output=str))
             options.append(define(
                 'Trilinos_EXTRA_LINK_FLAGS',
-                '-L%s/ -lgfortran' % (libgfortran),
+                '-L%s/ -lgfortran' % (libgfortran_dir),
             ))
 
         if sys.platform == 'darwin' and macos_version() >= Version('10.12'):

--- a/var/spack/repos/builtin/packages/trilinos/package.py
+++ b/var/spack/repos/builtin/packages/trilinos/package.py
@@ -682,15 +682,20 @@ class Trilinos(CMakePackage, CudaPackage):
         # ################# System-specific ######################
 
         # Fortran lib (assumes clang is built with gfortran!)
-        if ('+fortran' in spec and spec.compiler.name in ['gcc', 'clang', 'apple-clang']):
-            fc = Executable(spec['mpi'].mpifc) if ('+mpi' in spec) else Executable(spack_fc)
+        if ('+fortran' in spec
+                and spec.compiler.name in ['gcc', 'clang', 'apple-clang']):
+            fc = Executable(spec['mpi'].mpifc) if (
+                '+mpi' in spec) else Executable(spack_fc)
             sharedlibext = 'dylib' if sys.platform == 'darwin' else 'so'
             libgfortran = fc('--print-file-name',
-                'libgfortran.%s' % sharedlibext, output=str).strip()
+                             'libgfortran.%s' % sharedlibext,
+                             output=str).strip()
             # if libgfortran is equal to "libgfortran.<libext>" then
             # print-file-name failed, use static library instead
-            if (libgfortran == ('libgfortran.%s' %  sharedlibext)):
-                libgfortran = fc('--print-file-name', 'libgfortran.a', output=str).strip()
+            if (libgfortran == ('libgfortran.%s' % sharedlibext)):
+                libgfortran = fc('--print-file-name',
+                                 'libgfortran.a',
+                                 output=str).strip()
             options.append(define('Trilinos_EXTRA_LINK_FLAGS', libgfortran))
 
         if sys.platform == 'darwin' and macos_version() >= Version('10.12'):

